core/src/toga/widgets/multilinetextinput.py

```diff
@@ -1,148 +1,128 @@
+            the widget changes as a result of user input.
+        """
 
-        ######################################################################
-        # 2022-09: Backwards compatibility
-        ######################################################################
-        # factory no longer used
-        if factory:
-            warnings.warn("The factory argument is no longer used.", DeprecationWarning)
-        ######################################################################
-        # End backwards compatibility.
-        ######################################################################
+        super().__init__(id=id, style=style)
 
         # Create a platform specific implementation of a MultilineTextInput
         self._impl = self.factory.MultilineTextInput(interface=self)
 
-        ##################################################################
-        # 2022-07: Backwards compatibility
-        ##################################################################
-
-        # initial replaced with value
-        if initial is not None:
-            if value is not None:
-                raise ValueError(
-                    "Cannot specify both `initial` and `value`; "
-                    "`initial` has been deprecated, use `value`"
-                )
-            else:
-                warnings.warn("`initial` has been renamed `value`", DeprecationWarning)
-            value = initial
-
-        ##################################################################
-        # End backwards compatibility.
-        ##################################################################
+        # Set a dummy handler before installing the actual on_change, because we do not want
+        # on_change triggered by the initial value being set
+        self.on_change = None
+        self.value = value
 
         # Set all the properties
-        self.value = value
         self.readonly = readonly
         self.placeholder = placeholder
         self.on_change = on_change
 
     @property
     def placeholder(self):
-        """The placeholder text.
+        """The placeholder text for the widget.
 
-        Returns:
-            The placeholder text as a `str``.
+        A value of ``None`` will be interpreted and returned as an empty string.
+        Any other object will be converted to a string using ``str()``.
         """
-        return self._placeholder
+        return self._impl.get_placeholder()
 
     @placeholder.setter
     def placeholder(self, value):
-        self._placeholder = "" if value is None else str(value)
-        self._impl.set_placeholder(self._placeholder)
+        self._impl.set_placeholder("" if value is None else str(value))
+        self.refresh()
+
+    @property
+    def enabled(self):
+        """Is the widget currently enabled? i.e., can the user interact with the
+        widget?
+
+        Disabling a MultilineTextInput is equivalent to making the input
+        read-only.
```

Comment:
Disabling selection as a differentiator between disabled and readonly definitely makes sense (and will have a good analog with TextInput). I'm not sure disabling scrolling makes sense, though. I can't think of any example of a UI where a scroll bar *exists*, but can't be interacted with; from a quick poke around, I can't find any easily exposable API in Cocoa or iOS that would make that simple to implement.

Reply:
I've restored the inherited `enabled` handling for multilinetext, and made the Cocoa implementation a map to `selectable` (since Cocoa is the only platform that doesn't provide a native `enabled` API on multilinetext).